app/controllers/user_annotations_controller.rb

```diff
@@ -0,0 +1,154 @@
+class UserAnnotationsController < ApplicationController
+  before_action :set_user_annotation, only: [:show, :edit, :update, :destroy]
+  #Check that use is logged in in order to do anything
+  before_filter :authenticate_user!
+  before_action :check_permission, except: :index
+  # GET /user_annotations
+  # GET /user_annotations.json
+  def index
+    #get all this user's annotations
+    @user_annotations = current_user.user_annotations.owned_by(current_user)
+  end
+
+  # GET /user_annotations/1
+  # GET /user_annotations/1.json
+  def show
+  end
+
+
+  # GET /user_annotations/1/edit
+  def edit
+
+  end
+
+  # PATCH/PUT /user_annotations/1
+  # PATCH/PUT /user_annotations/1.json
+  def update
+    #update the annotation's defined labels
+    new_labels = user_annotation_params.to_h['values']
+    #If the labels sued to include undefined, make sure they do again
+    if @user_annotation.values.include? 'Undefined'
+      new_labels.push('Undefined')
+    end
+
+    #Remeber the old values
+    old_labels = @user_annotation.values
+
+    #Remeber the old annotations
+    annotation_arrays = @user_annotation.user_data_arrays.where(array_type: 'annotations').to_a
```

Comment:
You can call @user_annotation.user_data_arrays.by_name_and_type(name, array_type) instead.